src/client/public/scripts/components/row-element.ts

```diff
@@ -0,0 +1,85 @@
+import './row-header.js';
+
+import {html} from '../../../../../node_modules/lit-html/lib/lit-extended.js';
+import {TemplateResult} from '../../../../../node_modules/lit-html/lit-html.js';
+import {timeToString} from '../dash/utils/time-to-string.js';
+
+import {BaseElement, property} from './base-element.js';
+
+export interface StatusDisplay {
+  type: 'complete'|'actionable'|'activity'|'passive';
+  text: string;
+}
+
+export interface RowData {
+  id: string;
+  status: StatusDisplay;
+  createdAt: number;
+  author: string;
+  avatarUrl: string;
+  url: string;
+  title: string;
+  owner: string;
+  repo: string;
+  hasNewActivity: boolean;
+}
+
+export interface RowEvent {
+  text: string|TemplateResult;
+  time?: number;
+  url?: string;
+  classes?: string[];
+}
+
+export class RowElement extends BaseElement {
+  @property() data: RowData|null = null;
+  @property() extraHeaderData: TemplateResult[] = [];
+  @property() events: RowEvent[] = [];
+
+  _renderEvent(event: RowEvent) {
+    const timeTemplate = (time: number) =>
+        html`<time class="dashboard-row-event__time" datetime="${
+            new Date(time).toISOString()}">${timeToString(time)}</time>`;
+
+    const linkTemplate = (url: string, text: string|TemplateResult) =>
+        html`<a class="dashboard-row-event__url" href="${
+            url}" target="_blank">${text}</a>`;
+
+    return html`
+<div class$="dashboard-row-event ${
+        event.classes ? event.classes.join(' ') : ''}">
+  ${
+        event.time ? timeTemplate(event.time) :
```

Comment:
Could you move this if statement into the timeTemplate function?

Reply:
Done.